### DISCUSSION TITLE
RPM packages

### DISCUSSION THREAD

**Discussion by odeneriksson** [threathunters-io/laurel#229] [Ideas]
Hello,

I spent some time making RPM packages for RHEL that I hope will end up in RHEL somehow so I don't have to maintain this myself.

You can build binary RPM packages for RHEL 7, 8 & 9 using the single source rpm file. Should build for latest Fedora as well.

Please have a look at my repo that currently only have the source rpm here:

https://repo.nux.se/laurel/

Cheers,

**Comment by hillu** [comment]
Very nice, thank you. Should I add a link to the README – or should we add the spec file to contrib/ ?

**Reply by odeneriksson** [reply]
I'm doing some lobbying to make this end up in Fedora, then RHEL. You could add the spec file but it's not very pretty. The spec file would really need some massage by a skilled fedora rust developer, but it works good enough for now.

**Reply by odeneriksson** [reply]
I also added packages for Fedora 41 and RHEL 10 (beta).

**Reply by odeneriksson** [reply]
Oh, and lastly. The https://repo.nux.se/laurel/ link is a full blown yum repo with binary, debug and source rpm files. Well, you just wget either of these:

https://repo.nux.se/laurel/fedora/nux_laurel.repo
https://repo.nux.se/laurel/rhel/nux_laurel.repo

And yes, the packages has been built on RHEL and Fedora, not on any of the clones.

**Comment by odeneriksson** [comment]
Ok. After quite some lobbying at Redhat and pulling some strings there they concluded they will **NOT** take ownership of my prototype package and maintain it themselves.

So, the most sustainable way to make this happen is to do the work as a Fedora contributer and add it to Fedora and EPEL. Initially this will be a lot of work as every software in the "vendor" dependancy bundle needs to be packaged separately and follow the Fedora rules for rust packages.

There are some initial steps taken by my employer that will allow me to do this wearing that hat and during work hours. If so the packages may end up in Fedora and EPEL soon'ish.

**Comment by ec-cimpote** [comment]
Hey [odeneriksson](https://github.com/odeneriksson), we would like to take over maintaining a RPM package on code.europa.eu, mind you share your spec for it? (or just add it under contrib maybe)
